File: 2024/1/part_1.py
#!/usr/bin/python

file = open("input");
first_list = []
second_list = []
for line in file:
    fields = line.strip().split()
    first_list.append(int(fields[0]))
    second_list.append(int(fields[1]))

first_list = sorted(first_list)
second_list = sorted(second_list)

result = 0
for i in range (1000):
    result += abs(first_list[i] - second_list[i])

print(f"Result: {result}")

File: 2024/1/part_1.cpp
#include <algorithm>
#include <cstddef>
#include <fstream>
#include <iostream>
#include <stdexcept>
#include <vector>

int main()
{
    std::vector<int> firstList;
    std::vector<int> secondList;

    std::ifstream file("../2024/1/input");
    if (!file.is_open())
    {
        throw std::runtime_error("Can't open file!");
    }

    int number1, number2 = 0;
    while (file >> number1 >> number2)
    {
        firstList.push_back(number1);
        secondList.push_back(number2);
    }

    std::sort(firstList.begin(), firstList.end());
    std::sort(secondList.begin(), secondList.end());

    int result = 0;
    for (size_t i = 0; i < firstList.size(); i++)
    {
        result += abs(firstList[i] - secondList[i]);
    }

    std::cout << "Result: " << result << "\n";
    return 0;
}
